mllib/src/main/scala/org/apache/spark/mllib/optimization/GradientDescent.scala

```diff
@@ -122,8 +122,8 @@ class GradientDescent private[spark] (private var gradient: Gradient, private va
    * @return solution vector
    */
   @DeveloperApi
-  def optimize(data: RDD[(Double, Vector)], initialWeights: Vector): Vector = {
-    val (weights, _) = GradientDescent.runMiniBatchSGD(
+  def optimize(data: RDD[(Double, Vector)], initialWeights: Vector): (Vector, Double, Integer) = {
```

Comment:
There aren't currently; we need to find the time for it and might not during this next release---but we'll see!  If any, I'd suggest you watch [https://issues.apache.org/jira/browse/SPARK-5256]; it's outdated but will become active when we have the chance to work more on that API.  Thanks!